lib/Optimizer/CodeGen/LowerToQIR.cpp

```diff
@@ -138,6 +138,18 @@ class QmemRAIIOpRewrite
     // Get the CC Pointer for the state
     auto ccState = adaptor.getInitState();
 
+    // Inspect the element type of the complex data, need to
+    // know if its f32 or f64
+    Type eleTy = rewriter.getF64Type();
+    if (auto llvmPtrTy = dyn_cast<LLVM::LLVMPointerType>(ccState.getType())) {
+      auto ptrEleTy = llvmPtrTy.getElementType();
+      if (auto llvmStructTy = dyn_cast<LLVM::LLVMStructType>(ptrEleTy))
+        if (llvmStructTy.getBody().size() == 2 &&
+            llvmStructTy.getBody()[0] == llvmStructTy.getBody()[1] &&
+            llvmStructTy.getBody()[0].isF32())
+          eleTy = rewriter.getF32Type();
+    }
```

Comment:
Shouldn't this always be a `complex<f>*` type?  If so, we should probably add hard checks here. If we don't find that type, then raise an error on this Op as being broken.

Reply:
I'll update to throw an error